index.rst

```diff
@@ -5,77 +5,331 @@
+
+For example, one might be interested in a rule that activates when a new image finishes reading (camera publishes endReadout) or something a bit more complicated like activate when a new image finishes reading, the telescope is pointing in the direction of the wind and wind speed is above a certain threshold.
+
+In terms of design, we have two different options we could implement; configuration-based or programmatic-based rules.
+
+Configuration-based rules, are based on simple rules that can be combined together to form more complex behavior.
+For example, we could a rule that receives a component name and event name and is activated every time that event is published or a slightly more complicated rule that takes 2 components/topic/attributes and check if their values are within a certain threshold.
+
+The configuration for a rule that would monitor an event would look something like this:
+
+.. code-block:: yaml
+
+   rule:
+      monitorEvent:
+         component: ATCamera
+         event: endReadout
 
-One of the key features expected from the Catcher was the ability to issue alarms that can be displayed in LOVE.
-One option to support this feature would be to create a custom LOVE producer that can be interacted with using a REST API.
-The alert information would be persisted in Sasquatch by the `Flux`_ script, which would then post a message to the LOVE producer of the new alert.
-The producer would then read the alert information from the EFD and display it.
+Combining rules would be something like:
+
+.. code-block:: yaml
+
+   rule:
+      monitorEvent:
+         component: ATCamera
+         event: endReadout
+      compareTopics:
+         topic1: ATMCS.tel_mount_AzEl_Encoders.azimuthCalculatedAngle
+         topic2: WeatherStation.windDirection.value
+         threshold: 1.0
+
+The advantage of this approach is that, once a sufficient number of rules are developed, including new triggers is a matter of creating a new configuration.
+
+Alternatively, the programmatic-based rules relies on having all the trigger logic developed under a single operation.
+For example, we could have a rule to monitor the end readout event from the camera and a rule that monitors the end readout event from the camera and compare the position of the telescope with the wind.
+These rules are developed specifically for their use-cases, and are not generally applied.
+This is, for instance, the current approach implemented by the ``Watcher``.
+Any new alarm requires code changes. 
+
+Analysis Process
+----------------
+
+A key functionality of the ``Catcher``, that differentiates it from the ``Watcher``, is the ability to execute an analysis process once a certain condition is met.
+In general, we assume that these analysis will consist of lightweight computations.
+Whenever possible, the computation should be offloaded to the OCPS.
```

Comment:
I think it's only possible to offload it if it's image related... so maybe add that.